yesod-core/src/Yesod/Core/Content.hs

```diff
@@ -226,6 +240,53 @@ typeOctet = "application/octet-stream"
 simpleContentType :: ContentType -> ContentType
 simpleContentType = fst . B.break (== _semicolon)
 
+decoderForCharset :: Maybe B.ByteString -> L.ByteString -> TL.Text
+decoderForCharset (Just encodingSymbol)
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "utf-8")        = LE.decodeUtf8With EE.lenientDecode
```

Comment:
See the [IANA documentation](https://www.iana.org/assignments/character-sets/character-sets.xhtml) for the character set symbols. I didn't handle synonyms here.